addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -55,6 +87,159 @@ def _compute_priority(self):
             if new_priority != batch.priority:
                 batch.priority = new_priority
 
+    @api.multi
+    @api.constrains('user_id')
+    def _compute_state(self):
+        """ Compute the state of a batch post confirm
+            waiting     : At least some picks are not ready
+            ready       : All picks are in ready state (assigned)
+            in_progress : All picks are ready and a user has been assigned
+            done        : All picks are complete (in state done or cancel)
+
+            the other two states are draft and cancel are manual
+            to transitions from or to the respective state.
+        """
+        for batch in self:
+
+            if batch.state in ['draft', 'done', 'cancel']:
+                # Can not do anything with them don't bother trying
+                continue
+
+            ready_picks, other_picks = \
+                self._calculate_pick_groups(batch.picking_ids)
+
+            if not batch.picking_ids or not (ready_picks or other_picks):
+                # This accounts for all pickings being complete, canceled or
+                # all pickings get removed for some reason.
+                # The last two can happen in any state
+                batch.state = 'done'
+
+            # Attempt transitions based on state
+            elif batch.state == 'waiting':
+                batch._from_waiting_transitions(other_picks)
+
+            elif batch.state == 'ready':
+                batch._from_ready_transitions(other_picks)
+
+            elif batch.state == 'in_progress':
+                batch._from_in_progress_transitions(ready_picks, other_picks)
+
+            else:
+                # How did we get here?
+                # Let do nothing to be safe!
+                continue
+
+    def _from_in_progress_transitions(self, ready_picks=None, other_picks=None):
+        """ Transition from in_progress to another state"""
+        self.ensure_one()
+
+        # Make sure we have the pick groups
+        if ready_picks is None or other_picks is None:
+            ready_picks, other_picks = \
+                self._calculate_pick_groups(self.picking_ids)
+
+        # This transitions should be handled already in _compute_state
+        # but as a guard against it being called else where by someone
+        # lets add it in here as well
+        if not ready_picks and not other_picks:
+            # All picks are complete so we're done
+            self.state = 'done'
+            return True
+
+        elif ready_picks and not other_picks and not self.user_id:
+            # User is removed lets go back to being ready
+            self.state = 'ready'
+            return True
+
+        # None really a transition but a bit of batch pick managemnet
+        # which has to be delt with
```

Comment:
The comment should go inside the `elif` block. Please remove the above empty line, otherwise it's hard to identify the blocks and keep `if` & `elif`s associated.